.github/workflows/ci.yml

```diff
@@ -53,16 +61,20 @@ jobs:
           git diff --exit-code
 
   publish:
-    needs: [test-modules]
+    environment: ${{ github.event_name == 'release' && 'production' || '' }}
+    needs: [checks, test-modules]
     runs-on: ubuntu-latest
     permissions:
       contents: read
       id-token: write
     steps:
+      - run: true
       - uses: actions/checkout@v4
+        if: fromJson(env.PUBLISH_TO_FLAKEHUB)
```

Comment:
Why not put this on an `if:` on the entire job instead of each individual step...?

Reply:
So it still serves as a "did all the checks pass?" step, where otherwise it'd be skipped

Reply:
good idea! pused